src/main/java/com/mraof/minestuck/item/weapon/AspectBasedEffectWeaponItem.java

```diff
@@ -41,18 +33,25 @@ public AspectBasedEffectWeaponItem(IItemTier tier, int attackDamageIn, float att
     }
     
     @Override
-    public void inventoryTick(ItemStack stack, World worldIn, Entity entityIn, int itemSlot, boolean isSelected)
+    public boolean hitEntity(ItemStack itemStack, LivingEntity target, LivingEntity attacker)
     {
-        if(isSelected && entityIn instanceof ServerPlayerEntity)
+        Title title = PlayerSavedData.getData((ServerPlayerEntity) attacker).getTitle();
+        
+        if(attacker instanceof ServerPlayerEntity && attacker.getRNG().nextFloat() < .1)
         {
-            Title title = PlayerSavedData.getData((ServerPlayerEntity) entityIn).getTitle();
             if(title != null)
             {
                 if (title.getHeroAspect() == aspect)
                 {
-                    ((ServerPlayerEntity) entityIn).addPotionEffect(new EffectInstance(effect.get(), duration, effectTier));
+                    ((ServerPlayerEntity) attacker).addPotionEffect(new EffectInstance(effect.get(), duration, effectTier));
```

Comment:
Got a suggestion to have an EffectInstance supplier instead of an Effect supplier to bundle together the effect parameters, but feel free to leave it as it is

Reply:
Would that mean that the original code should also be changed?

Reply:
Yeah

Reply:
I am unsure what the change would look like, if you type it out here I can include it with my other changes